### PR TITLE
[corefx] Import System.IO.MemoryMappedFiles

### PR DESCRIPTION
**Pull request by alexischr**
Implements #6975

### CODE REVIEW

File: mcs/class/System.Core/common_System.Core.dll.sources [hunk 1]
```diff
@@ -3,10 +3,14 @@
 
 Assembly/AssemblyInfo.cs
 System/Util.cs
-System.IO.MemoryMappedFiles/MemoryMappedFile.cs
-System.IO.MemoryMappedFiles/MemoryMappedView.cs
-Microsoft.Win32.SafeHandles/SafeMemoryMappedFileHandle.cs
-Microsoft.Win32.SafeHandles/SafeMemoryMappedViewHandle.cs
+../../../external/corefx//src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFileRights.cs
+../../../external/corefx//src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFileAccess.cs
```
Comment (marek-safar) [inline mcs/class/System.Core/common_System.Core.dll.sources]: nit: extra /

File: mcs/class/System.Core/corefx.unix.sources [hunk 2]
```diff
@@ -0,0 +1,18 @@
+../../../external/corefx/src/Common/src/Interop/Unix/*.cs
+../../../external/corefx//src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedView.Unix.cs
```
Comment (marek-safar) [inline mcs/class/System.Core/corefx.unix.sources]: nit: extra /

**Comment by luhenry** [comment]
@alexischr you should be able to rebase on top of master to catch up on the System.Native integration.

**Comment by alexischr** [comment]
@monojenkins build failed

**Comment by steveisok** [comment]
Since this is older, I'm closing.  Please see the [old pull request policy](https://github.com/mono/mono/wiki/Pull-Request-Guide#old-pull-request-policy) for more info.

**Pull request closed without merging**